src/core/Store.ts

```diff
@@ -0,0 +1,40 @@
+import { set } from "../utils/set";
+import EventBus from "./EventBus";
+import { IState } from "../api/types";
+
+export enum StoreEvents {
+  Updated = 'updated'
+}
+
+export class Store extends EventBus {
+  private state: IState = {
```

Comment:
Было бы не плохо сделать Store типизируемым через дженерики, вот так
```suggestion
export class Store<TState> extends EventBus {
  private state: TState
```